symengine/as_real_imag.cpp

```diff
@@ -0,0 +1,284 @@
+#include <symengine/visitor.h>
+#include <symengine/basic.h>
+
+namespace SymEngine
+{
+
+class RealImagVisitor : public BaseVisitor<RealImagVisitor>
+{
+private:
+    Ptr<RCP<const Basic>> real_, imag_;
```

Comment:
I thought it would be easier with https://github.com/symengine/symengine/pull/1310#discussion_r127377358, but I'm okay either way